types/datetime.go

```diff
@@ -0,0 +1,292 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+	"unicode"
+)
+
+type Datetime struct {
+	// Value is a timestamp in milliseconds
+	Value int64
+}
+
+func UnsafeDatetime[T int | int64 | float64](v T) Datetime {
+	return Datetime{Value: int64(v)}
+}
+
+func FromStdTime(t time.Time) Datetime {
```

Comment:
Maybe `DatetimeFromStandardTime` ... idk.

Reply:
For Duration, I did `FromStdDuration` -- paired together, they make sense. We're lifting the "standard type" into Cedar Values. Could do the same for net.IP -- `FromStdIP`. 

I'm also flexible so don't really care one way or the other. :)